Sources/Commands/SwiftTool.swift

```diff
@@ -794,8 +793,8 @@ extension BuildSubset {
 
 /// Returns path of the nearest directory containing the manifest file w.r.t
 /// current working directory.
-private func findPackageRoot() -> AbsolutePath? {
-    guard var root = localFileSystem.currentWorkingDirectory else {
+private func findPackageRoot(for packagePath: AbsolutePath?) -> AbsolutePath? {
+    guard var root = packagePath else {
```

Comment:
Here, we could use the resolved `packagePath` variable (mentioned in the comment above) instead of taking it as argument.